crates/turbopack/src/module_options/mod.rs

```diff
@@ -99,23 +100,13 @@ impl ModuleOptionsVc {
         if enable_styled_jsx {
             transforms.push(EcmascriptInputTransform::StyledJsx);
         }
-        if let Some(enable_emotion) = enable_emotion {
-            let emotion_transform = enable_emotion.await?;
-            transforms.push(EcmascriptInputTransform::Emotion {
-                sourcemap: emotion_transform.sourcemap.unwrap_or(false),
-                label_format: OptionStringVc::cell(emotion_transform.label_format.clone()),
-                auto_label: if let Some(auto_label) = emotion_transform.auto_label.as_ref() {
-                    match auto_label {
-                        EmotionLabelKind::Always => Some(true),
-                        EmotionLabelKind::Never => Some(false),
-                        // [TODO]: this is not correct coerece, need to be fixed
-                        EmotionLabelKind::DevOnly => None,
-                    }
-                } else {
-                    None
-                },
-            });
+
+        if let Some(transformer) = build_emotion_transformer(enable_emotion).await? {
+            transforms.push(EcmascriptInputTransform::Plugin(TransformPluginVc::cell(
+                transformer,
+            )));
         }
```

Comment:
(Eventually) `enable_emotion` should be removed from the ModuleOptionsContext and passed as `custom_ecmascript_transforms`

Reply:
Yes, that makes more sense.